modules/tasks/tasks/src/main/java/org/motechproject/tasks/service/impl/TaskServiceImpl.java

```diff
@@ -579,6 +581,22 @@ private TaskDataProvider findProviderByName(DataSource ds) {
         return errors;
     }
 
+    private void validateName(Task task) {
+
+        Long taskId = task.getId();
+        int tasksWithName = tasksDataService.findTasksByName(task.getName()).size();
```

Comment:
Can we store the returned objects here, so we don't have to repeat database call in line 593?